src/dev.rs

```diff
@@ -0,0 +1,293 @@
+//! Tools for developing circuits.
+
+use ff::Field;
+
+use crate::{
+    arithmetic::{FieldExt, Group},
+    plonk::{Any, Assignment, Circuit, Column, ConstraintSystem, Error},
+    poly::{EvaluationDomain, LagrangeCoeff, Polynomial},
+};
+
+#[derive(Debug)]
+struct Cell(usize, usize);
+
+/// The reasons why a particular circuit is not satisfied.
```

Comment:
Maybe we should move this out of dev so that we can more precisely report unsatisfiedness (dissatisfaction?) errors in the normal case. I think it's always useful to do that even if it is more expensive, since unsatisfiedness indicates a serious bug in either the circuit implementation or the provider of the high-level witness.

Reply:
We obviously can't use this for normal verification, because we don't have any of the witness information 🙂

We could possibly include this within the prover's errors (by caching more information during the proof-creation process, to enable this granularity), but in general it would be better to encourage `Circuit` structs to be designed in a way that makes it impossible to construct invalid witness assignments.

An alternative we could explore is that if creation of a proof fails, we immediately run that circuit assignment through the `MockProver` to get the more granular details.

Reply:
Opened #99 for this.